### PR TITLE
Adding role based filtering for Preview API

### PR DESCRIPTION
**Pull request by saratvemulapalli**
*Issue #195 *

*Description of changes:*
Added backend role based filtering for Preview API.
Added new integration tests to verify the behavior.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: src/main/java/com/amazon/opendistroforelasticsearch/ad/transport/PreviewAnomalyDetectorTransportAction.java [hunk 1]
```diff
@@ -67,16 +74,39 @@ public PreviewAnomalyDetectorTransportAction(
...
         this.xContentRegistry = xContentRegistry;
         maxAnomalyFeatures = MAX_ANOMALY_FEATURES.get(settings);
         clusterService.getClusterSettings().addSettingsUpdateConsumer(MAX_ANOMALY_FEATURES, it -> maxAnomalyFeatures = it);
+        filterByEnabled = AnomalyDetectorSettings.FILTER_BY_BACKEND_ROLES.get(settings);
+        clusterService.getClusterSettings().addSettingsUpdateConsumer(FILTER_BY_BACKEND_ROLES, it -> filterByEnabled = it);
     }
 
     @Override
     protected void doExecute(Task task, PreviewAnomalyDetectorRequest request, ActionListener<PreviewAnomalyDetectorResponse> listener) {
+        String detectorId = request.getDetectorId();
```
Comment (ylwu-amzn) [inline src/main/java/com/amazon/opendistroforelasticsearch/ad/transport/PreviewAnomalyDetectorTransportAction.java:89]: Detector id could be null. No need to call `resolveUserAndExecute` if detector id is null.

Comment (saratvemulapalli) [reply src/main/java/com/amazon/opendistroforelasticsearch/ad/transport/PreviewAnomalyDetectorTransportAction.java:89]: Thats a good point, sure.

**Review by kaituo — APPROVED** [review]

File: src/main/java/com/amazon/opendistroforelasticsearch/ad/transport/PreviewAnomalyDetectorTransportAction.java (hunk 1 above)
Comment (ylwu-amzn) [reply src/main/java/com/amazon/opendistroforelasticsearch/ad/transport/PreviewAnomalyDetectorTransportAction.java:89]: Sorry, I though the detector id was moved to request body. As we discussed, the detector id id part of request URL, we can suppose it always not null. You can keep the code as is.

**Review by ylwu-amzn — APPROVED** [review]

File: src/main/java/com/amazon/opendistroforelasticsearch/ad/transport/PreviewAnomalyDetectorTransportAction.java (hunk 1 above)
Comment (saratvemulapalli) [reply src/main/java/com/amazon/opendistroforelasticsearch/ad/transport/PreviewAnomalyDetectorTransportAction.java:89]: Had a quick chat with @ylwu-amzn. 
Detector ID will not be null its its part of the request path. 
Will leave this as it is to cover a case where user puts in a different detector ID which he does not own, we will be rejecting the request.

Comment (ylwu-amzn) [reply src/main/java/com/amazon/opendistroforelasticsearch/ad/transport/PreviewAnomalyDetectorTransportAction.java:89]: Currently we have 2 steps: 
1.Configure data source basic info like index, interval, window delay, then save it to index, so we will get detector id 
2.Configure features, user can call preview API before feature changes saved to index.

In future, we may change to 1 step, configure data source basic info and feature on the same page, then user can call preview API before saving the detector to index, so we will not get detector id. Then we should move detector id into request body. We can refactor code when we make the change from frontend.

Comment (saratvemulapalli) [reply src/main/java/com/amazon/opendistroforelasticsearch/ad/transport/PreviewAnomalyDetectorTransportAction.java:89]: Sure makes sense. Thats probably better for the customer.

**Pull request merged**